### ISSUE TITLE
Add language to the Fetch query endpoint

### ISSUE THREAD

**Issue by infojunkie** [meedan/check#15]
Allow searching for fact-checks based on their language.  Add a `language` parameter to the `claim_reviews` endpoint that accepts a string, and defaults to none. 

- Match the `language` field of the Elasticsearch documents, by adjusting [the ES query](https://github.com/meedan/fetch/blob/develop/lib/elastic_search_query.rb#L104-L112) appropriately.
- Add a [test for the `language` parameter on the `claim_review` endpoint](https://github.com/meedan/fetch/blob/develop/spec/models/claim_review_test.rb)
- Add a [test for the `language` Elasticsearch query](https://github.com/meedan/fetch/blob/develop/spec/lib/elastic_search_query_test.rb)
- Ensure [all tests pass locally](https://github.com/meedan/fetch#testing)

**Comment by DGaffney** [comment]
As it is currently written, we have an endpoint that essentially does this already - `GET /claim_reviews.json` has the following structure:
```
      {
        method: 'GET',
        params: {
          query: 'string (default none)',
          service: 'string (default none)',
          start_time: "time-parseable string (e.g. '2020-01-01' or 'Sept 20 2019', default none)",
          end_time: "time-parseable string (e.g. '2020-01-01' or 'Sept 20 2019', default none)",
          per_page: 'integer (default 20)',
          offset: 'integer (default 0)'
        }
      }
```

Here, we can cursor through time using the two `time` parameters, specify a `provider` using the `service` option, and specify a `query` with `query`, which should work fine with any valid ES-type logic query in that spot. The only one not present in here is `languageCode` which I agree would likely be useful. If we do this, the only thing that needs to happen is adding this `language` field to the query.

**Comment by khushijindal** [comment]
Hey @infojunkie! Can I work on this issue?

**Comment by infojunkie** [comment]
> Hey @infojunkie! Can I work on this issue?

Please!

**Comment by khushijindal** [comment]
Hey, @DGaffney can you help me how to proceed as you have mentioned, I am not able to find the file with the given code. I am new to open-source, so your help will be valuable.

**Comment by DGaffney** [comment]
> Hey, @DGaffney can you help me how to proceed as you have mentioned, I am not able to find the file with the given code. I am new to open-source, so your help will be valuable.

Hi! Glad to have you here. The code that I'm referencing will mostly need to be added over in the [fetch repository](https://github.com/meedan/fetch). Specifically, in the `lib/elastic_search_query.rb` file around line 108, a new conditional clause would need to be added that'd look something like:

```
    if opts[:language]
      query[:query][:bool][:filter] << ElasticSearchQuery.query_match_clause('language', opts[:language])
    end
```

This would also probably need a few new tests as well. Does this help clarify? Happy to discuss more!

**Comment by khushijindal** [comment]
I would need to add new tests in the fetch/spec/lib/elastic_search_query_test.rb and fetch/spec/models/claim_review_test.rb files I suppose. @DGaffney

**Comment by khushijindal** [comment]
https://github.com/meedan/fetch/pull/64
https://github.com/meedan/fetch/pull/65

**Comment by danielafeitosa** [comment]
Closed with meedan/fetch#64 and meedan/fetch#65
